matrix_sdk_base/src/models/room.rs

```diff
@@ -18,32 +18,51 @@ use std::collections::{BTreeMap, HashMap};
 use std::convert::TryFrom;
 
 #[cfg(feature = "messages")]
-use super::message::MessageQueue;
+use super::message::{MessageQueue, MessageWrapper};
 use super::RoomMember;
 
 use crate::api::r0::sync::sync_events::{RoomSummary, UnreadNotificationsCount};
-use crate::events::collections::all::{RoomEvent, StateEvent};
 use crate::events::presence::PresenceEvent;
 use crate::events::room::{
-    aliases::AliasesEvent,
-    canonical_alias::CanonicalAliasEvent,
-    encryption::EncryptionEvent,
-    member::{MemberEvent, MembershipChange},
-    name::NameEvent,
-    power_levels::{NotificationPowerLevels, PowerLevelsEvent, PowerLevelsEventContent},
-    tombstone::TombstoneEvent,
+    aliases::AliasesEventContent,
+    canonical_alias::CanonicalAliasEventContent,
+    encryption::EncryptionEventContent,
+    member::{MemberEventContent, MembershipChange},
+    name::NameEventContent,
+    power_levels::{NotificationPowerLevels, PowerLevelsEventContent},
+    tombstone::TombstoneEventContent,
+};
+
+use crate::events::{
+    Algorithm, AnyMessageEventContent, AnyRoomEventStub, AnyStateEventStub,
+    AnyStrippedStateEventStub, EventJson, EventType, StateEventStub, StrippedStateEventStub,
 };
-use crate::events::stripped::{AnyStrippedStateEvent, StrippedRoomName};
-use crate::events::{Algorithm, EventType};
 
 #[cfg(feature = "messages")]
-use crate::events::room::message::MessageEvent;
+use crate::events::{
+    room::redaction::{RedactionEvent, RedactionEventStub},
+    AnyMessageEventStub,
+};
 
 use crate::identifiers::{RoomAliasId, RoomId, UserId};
 
 use crate::js_int::{Int, UInt};
 use serde::{Deserialize, Serialize};
-#[derive(Debug, Default, PartialEq, Serialize, Deserialize, Clone)]
+
+#[cfg(feature = "messages")]
+fn full_event_from_stub(event: RedactionEventStub, room_id: RoomId) -> RedactionEvent {
```

Comment:
Can we get a better name for this, it sounds like it can convert any stub event to a fill event, but it can do so only for redaction events.